### PR TITLE
feat: implemented ifftshift and corresponding test

### LINKED ISSUE #23801: ifftshift

### PR DESCRIPTION
**Pull request by VaishnaviMudaliar**
# PR Description

## Related Issue

Closes #23801 

## Checklist

- [X] Did you add a function?
- [X] Did you add the tests?
- [X] Did you run your tests and are your tests passing?
- [X] Did pre-commit not fail on any check?
- [X] Did you follow the steps we provided?

### CODE REVIEW

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi Thanks for the PR.
It seems like test_jax_numpy_ifftshift is still failing as seen in the logs here:
image

**Comment by VaishnaviMudaliar** [comment]
image


@Ishticode Kindly review the code , all tests are passing now , however , if anymore changes are needed , kindly let me know 👍

**Comment by VaishnaviMudaliar** [comment]
image
Pre commit passing too.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py [hunk 1]
```diff
@@ -271,6 +271,30 @@ def test_jax_numpy_ifftn(
...
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        backend_to_test=backend_fw,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        test_values=True,
+        x=arr[0],
+        axes=None,  # You can change this to test specific axes if needed
```
Comment (Ishticode) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py:294]: This should be auto tested as in auto generated and tested for. I think there is a helper like `dtype_values_axis`
Passing a None as the arg value is testing it only for that particular case always.

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi Thanks for your patience @VaishnaviMudaliar 
Below is a request

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py (hunk 1 above)
Comment (VaishnaviMudaliar) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py]: I have made the required changes , kindly have a look

**Comment by VaishnaviMudaliar** [comment]
image
I have generated the axis parameter as suggested by you by using dtype_values_axis helper function. The tests are passing. Kindly let me know if anymore changes are needed 👍

**Review by Ishticode — APPROVED** [review]
Hi @VaishnaviMudaliar 
This looks good now. Thank you very much for your effort and patience through this. Will merge this soon. 🙂

**Pull request merged**